design/Proposal For View Definition.md

```diff
@@ -0,0 +1,84 @@
+﻿
+
+
+
+## Proposal to extend the View Definition
+**Background**
+The current  interface for a SqlView  which is a first class object in Nessie specifies a Sql text and a Dialect  and schema :
+
+    public interface SqlView extends Contents {  
+      
+      public static enum Dialect {  
+        HIVE,  
+        SPARK,  
+        DREMIO,  
+        PRESTO  
+      }  
+      
+      String getSqlText();  
+      
+      Dialect getDialect();  
+      
+      // Schema getSchema();  
+    }
+
+While thinking of integrating Dremio with Nessie, we found we needed several more fields to fully define a view along with  a schema  (written up in a separate proposal) 
+
+**Proposal for additional fields**
+While a View is ultimately describe by just the SQL query , these  additional fields needed to describe a view (or what we refer to as a Virtual Dataset in the context of Dremio) for the DREMIO dialect . 
```

Comment:
What makes the DREMIO dialect special in that regard?

Reply:
One difference may be how a specific version of a table is referred to. For e.g. a View can refer to a specific version of a table in a specific branch. This particular syntax could be different in different dialects